### DISCUSSION TITLE
Running MC in cloud

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5547] [Q&A]
We currently use mirth appliances to run mirth connect. We have toyed with the idea of moving mirth to the cloud but I have a few questions on how certain things would work there.

1. In the appliance, it is very easy to apply updates as they become available as it does all the work for you.  How is a standalone mirth connect updated in the cloud? Do you have to install the new version and export\import all of your channels over to the new version?

2. How are VPN connections handled in the cloud? Again, easy to do with the appliance.

3. Generally, are there any big advantages (or disadvantages) to running mirth in the cloud vs maintaining the current setup of appliance\mc?

**Comment by pacmano1** [comment]
"Moving to the cloud" is a complex question.   For that matter "moving appliances to the cloud" is a complex question.   

Honestly, if this is something you are considering and you are not deeply familair with AWS, Azure, etc. you should engege a senior resource to help.    You can't really "toy" with the concept.   As for upgrading a standalone non-mirth appliance, that is easy enough cloud or not.

Your questions involves best practice topology design considerations in the cloud irrespective of mirth.   For example, Mirth would typically sit in AWS in a private subnet of a VPC, and if it served public https endpoints those would be front ended by an application load balancer and WAF that sits in a public subnet of the VPC, possibly firewalled to allow inbound connections from certain ports.    Your acess to Mirth might be via a corporate IPSec tunnel or MFA VPN to that private subnet in the VPC.     Your clients' TCP based connections might come in over IPSec connections.   There is much to consider.  VPNs might be provisioned via your own appliance in the cloud or the cloud provider's native VPN capabilities.

The advantages to cloud deployments are extensive - least of which is the fact that your company can't possibly compete with the dollars AWS, Azure, etc. to make their systems reliable, scalable, automatable and extensible.

**Reply by rutledj** [reply]
Thanks for the reply. According to Mirth support, the appliance can't operate in the cloud. I did read their cloud deployment guide referring to using the load balancer with multiple mirth instances (which we don't have now). I was more interested in the routine maintenance\updates aspect in the cloud.

**Reply by pacmano1** [reply]
I wouldn't use the appliance anyway.  

Routine maintenance is simple in my opinion.

**Reply by jonbartels** [reply]
https://aws.amazon.com/marketplace/seller-profile?id=893b7e01-47a8-4b55-b327-99edb31a7324 may be relevant here.

Also vendors like https://consultzen.com/ and ordinary users are clearly running Mirth Connect in the cloud.

The appliances will run in the cloud, just as VMs on EC2 instances rather than as any sort of cloud-native service. I don't know if that counts. 

Are you looking to DIY or would hiring a consultant to help migrate be valuable? https://diridium.com/ has specific expertise in this area I think.

**Reply by rutledj** [reply]
Thanks again for the information. Not entirely sure we will be moving to the cloud but just looking at that possibility. Our company was recently merged into another company and our previous parent company is hosting our appliances in their network. It will eventually have to be moved to our new company's infrastructure. Not sure yet what that will look like, so exploring possibilities.
